data/src/codes.py

```diff
@@ -8,182 +8,193 @@
+    chunk_size = 50
+    for start in range(0, len(categories), chunk_size):
+        end = start + chunk_size
+        requests_params = {
+            "action": "query",
+            "format": "json",
+            "prop": "categoryinfo",
+            "titles": "|".join(categories[start:end]),
+        }
+        data = requests.get(URL, params=requests_params).json()
+        for page in data["query"]["pages"].values():
+            size = page["categoryinfo"]["size"]
+            language = re.search(
+                r"Category:(.+?) terms with IPA pronunciation", page["title"]
+            ).group(1)
+            language_sizes[language] = size
+    return language_sizes
+
+
+def _scrape_wiktionary_language_code(lang_title: str) -> str:
+    """Extracts Wiktionary language code from language page."""
+    # Perhaps a bit overdetermined.
+    lang_code_selector = """
+    //tbody
+        /tr[@class = "language-category-data"]
+            /following-sibling::tr[
+                th
+                    /a[@title = "Wiktionary:Languages"]
+            ]//td
+                /code
+    """
 
-def _scrape_wiktionary_info(lang_title: str) -> Tuple[str, str]:
-    """Extracts Wiktionary language name and code from language page."""
-    name = ""
-    code = ""
     session = requests_html.HTMLSession()
     language_page = session.get(
-        f"https://en.wiktionary.org/wiki/{lang_title}_language", timeout=10
-    )
-    lang_table = language_page.html.find(
-        ".language-category-info > tbody > tr"
+        f"https://en.wiktionary.org/wiki/Category:{lang_title}_language",
+        timeout=10
     )
-    i = 0
-    while i < len(lang_table):
-        if "Canonical name" in lang_table[i].text:
-            name = lang_table[i].text
-            name = name.split("\n")[1]
-        elif "Language code" in lang_table[i].text:
-            # Canonical name (should) always be filled already.
-            # We grab Wiktionary language code as our entry point
-            # to compare/confirm with ISO 639 TSV File because
-            # Wiktionary name does not always correspond with
-            # ISO language name.
-            # Wiktionary: Ancient Greek; ISO: Greek, Ancient (to 1453).
-            code = lang_table[i].text
-            code = code.split("\n")[1]
-            break
-        i += 1
-    return (name, code)
+    return language_page.html.xpath(lang_code_selector)[0].text
+
+
+def _check_language_code_against_wiki(
+    language_code: str, language: str
+) -> None:
+    """Checks if WikiPron can handle the assigned ISO language code."""
+    try:
+        language_inferred = wikipron.Config(key=language_code).language
+    except iso639.NonExistentLanguageError:
+        logging.warning('WikiPron cannot handle "%s"', language)
+    else:
+        if language_inferred != language:
+            logging.warning(
+                'WikiPron resolves the key "%s" to "%s"'
+                'which is listed as "%s" on Wiktionary.',
```

Comment:
Small nit but a quick search said most logging statements here don't end in '.', maybe use that.

Also instead of typing "%s" in a logging statement, use %r, which is the same thing. Then you can use double-quotes for the actual logging string, like the rest of the file.